server/src/main/java/org/opensearch/index/translog/InternalTranslogManager.java

```diff
@@ -288,7 +312,19 @@ public void skipTranslogRecovery() {
         pendingTranslogRecovery.set(false); // we are good - now we can commit
     }
 
-    private Translog openTranslog(
+    // visible for testing
+    public Translog getTranslog() {
```

Comment:
Can we change it to package private instead of public?

Reply:
InternalEngineTests(outside this package) requires this and would need quite a refactor of tests

Reply:
I think you could improve the design by introducing `TranslogProvider` and passing it to `InternalTranslogManager`: in this case you could always reference `Translog` independently from `InternalTranslogManager` and drop this method (to be fair, if method is `public`, it is visible everywhere, ....)

Reply:
Sorry if I misunderstood, but the whole point of this change was to let TranslogManager manage the lifecycle of the Translog.

TranslogManger initialises performs all operations and closes the underlying Translog resource once done which kinda simplifies the overall design

 I however do plan on decoupling tests as well but that might take sometime and will follow it up

Reply:
Got it, the confusion for me is: I see a large number of methods in `InternalTranslogManager` delegating directly to `Translog` 1:1, it hints that `Translog` should not be exposed. From other side, there is a public `getTranslog()` method which basically defeats the purpose of delegation: why not to call the same methods using `getTranslog()`? 

I think it is fine to keep  `getTranslog()` and remove proxy methods, or introduce another abstraction (like provider) which could open the access to `Translog` in tests.

Reply:
So far what has been ensured is all delegation from the non-test classes are via the TranslogManager and only tests have the public method exposed which is temporarily done as it wasnt easy to get rid of the Translog dependency on tests unless we sufficiently refactor them.
At this point I would restrict all non-test and will follow it up..will mark a TODO for now. Let me know if that works

Reply:
Well, I don't see simple solution right now sadly, TODO would make sense for later, thanks @Bukhtawar